README.md

```diff
@@ -0,0 +1,79 @@
+# SmallRye Dux
+
+SmallRye Dux, Dux in short, is an extensible Service Discovery and Client-Side 
+Load Balancer implementation designed to work well with reactive libraries.
+Dux APIs use [SmallRye Mutiny](https://smallrye.io/smallrye-mutiny/).
+
+Dux maps service names to addresses. 
+
+Required `ServiceDiscovery` provides a list of addresses (`ServiceInstance`'s)
+for a given service name.
+Optional `LoadBalancer` selects a single `ServiceInstance` for a call.  
+
+## Configuring user applications
+The project's core aims to be vendor neutral. Aside from using Mutiny for the APIs,
+it has no dependencies. As such, it provides a universal abstraction of configuration 
+and vendors can choose their own way of providing Dux configuration.
+
+MicroProfile based frameworks can use MicroProfile Config to configure Dux. 
+Below is an example snippet of such a configuration:
+```properties
+dux.my-service.service-discovery=consul
+dux.my-service.service-discovery.some-service-discovery-property=property-value
+dux.my-service.load-balancer=round-robin
+dux.my-service.load-balancer.some-load-balancer-property=some-prop-value
+
+dux.my-other-service.service-discovery=test-sd-1
+dux.my-other-service.load-balancer=test-lb-1
+```
+
+MicroProfile Config support for configuring SmallRye Dux is provided by the 
+`smallrye-dux-microprofile` artifact.
+
+## Integrating client libraries with Dux
+Client libraries would most often use a `LoadBalancer` to get a single address
+for a call.
+`LoadBalancer` for a service can be retrieved from `Dux` singleton:
+
+```java
+LoadBalancer loadBalancer = Dux.getInstance().getLoadBalancer(serviceName);
```

Comment:
Did you think of CDI integration? Being able to `@Inject @ServiceName("my-service") LoadBalancer` (or `@Inject @ServiceName("my-service") ServiceDiscovery` would be nice.

Reply:
That should be easy to do :)

The classes that use it would usually be quite low-level, I didn't think it's likely they would be CDI beans.